raw-ipa-lib/src/helpers/event.rs

```diff
@@ -92,3 +109,88 @@ impl DerefMut for Helper {
         &mut self.public
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{Helper, Role};
+    use crate::user::User;
+    use crate::user::EventReport;
+        use crate::threshold::{
+        DecryptionKey as ThresholdDecryptionKey, EncryptionKey as ThresholdEncryptionKey,
+    };
+    pub use rust_elgamal::{Ciphertext, EncryptionKey, RistrettoPoint, Scalar, GENERATOR_POINT};
+    use rand::thread_rng;
+
+    #[test]
+    fn test_the_basics() {
+        let h_se = Helper::new(Role::Source);
+        let h_te = Helper::new(Role::Trigger);
+
+        let tek = ThresholdEncryptionKey::new(&[h_se.public.matchkey_encrypt, h_te.public.matchkey_encrypt]);
+
+        const PROVIDER_1: &str = "social.example";
+        const PROVIDER_2: &str = "news.example";
+        const PROVIDER_3: &str = "email.example";
+        const PROVIDER_4: &str = "game.example";
+
+        const MATCHING_MATCHKEY: &str = "12345678";
+
+        let mut u123 = User::new(123, tek);
+        u123.set_matchkey(PROVIDER_1, MATCHING_MATCHKEY);
+        u123.set_matchkey(PROVIDER_2, "23456789");
+        u123.set_matchkey(PROVIDER_3, "34567890");
+        u123.set_matchkey(PROVIDER_4, "45678901");
+
+        let mut u234 = User::new(234, tek);
+        u234.set_matchkey(PROVIDER_1, MATCHING_MATCHKEY);
+        u234.set_matchkey(PROVIDER_2, "does_not_match");
+        u234.set_matchkey(PROVIDER_3, "something_else");
+
+        let r123 = u123.generate_event_report(&[PROVIDER_1, PROVIDER_2, PROVIDER_3, PROVIDER_4]);
+        let r234 = u234.generate_event_report(&[PROVIDER_1, PROVIDER_2, PROVIDER_3]);
+
```

Comment:
Some checks for equality here are probably useful.  Maybe you can just create a helper function for doing that.